server/controllers/admin/campaign/getDonorsByCampaignId.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response, NextFunction } from 'express';
+import { Campaign, Donor } from '../../../database/models';
+import { CustomedError } from '../../../utils';
+import { paramsSchema } from '../../../utils/validation';
+
+const getDonorsByCampaignId = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const {
+      page = 1,
+    }: any = req.query;
+
+    const { id } = await paramsSchema.validateAsync(req.params);
+    const donors = await Campaign.findByPk(id, {
```

Comment:
it is better to use function find and count  it is  return the count of campaigns  that you need in  pagination

Reply:
https://sequelize.org/docs/v6/core-concepts/model-querying-finders/#findandcountall